### PR TITLE
Added Layer and Layered surface property

### LINKED ISSUE #1310: Create Layered class of ISurfaceProperty for layered shells
#### Description:

Similar to a Physical_oM Construction, but intended for analytical panels in order to send to FE packages accepting layered shells, and also mapping directly to Physical Constructions or Revit layered panel properties.

### PR DESCRIPTION
**Pull request by JosefTaylor**
### Issues addressed by this PR

Closes #1310

### CODE REVIEW

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check versioning

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
Small tweak on one of the properties, and some comments on one of the descriptions. Appart from that, I think this is looking good!

File: Structure_oM/SurfaceProperties/Layer.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        [Description("The material this layer is made of.")]
+        public virtual IMaterialFragment Material { get; set; }
+
+        [Angle]
+        [Description("The orientation of the material in this layer.")]
+        public virtual Angle Orientation { get; set; } = null;
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_oM/SurfaceProperties/Layer.cs:43]: Good catch, and I think that's the right idea making the definition a bit more explicit.

**Comment by JosefTaylor** [comment]
@BHoMBot check installer

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by JosefTaylor** [comment]
@BHoMBot check copyright-compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check ready-to-merge

**Review by IsakNaslundBh — DISMISSED** [review]
Happy with this!

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot fix copyright headers ref. 4840733055

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Review by IsakNaslundBh — APPROVED** [review]
Re-approving after Copyright updates

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**